src/domain/individualHarbor/forms/berthForm/BerthForm.tsx

```diff
@@ -0,0 +1,192 @@
+import React from 'react';
+import { Formik } from 'formik';
+import * as Yup from 'yup';
+import { useTranslation } from 'react-i18next';
+import { TFunction } from 'i18next';
+import { Button, TextInput } from 'hds-react';
+import { ObjectSchema } from 'yup';
+
+import { Berth, FormProps } from './types';
+import { BerthMooringType } from '../../../../@types/__generated__/globalTypes';
+import Grid from '../../../../common/grid/Grid';
+import Select from '../../../../common/select/Select';
+import Checkbox from '../../../../common/checkbox/Checkbox';
+import styles from './berthForm.module.scss';
+import { Pier } from '../../utils/utils';
+import Text from '../../../../common/text/Text';
+
+interface BerthFormProps extends FormProps<Berth> {
+  onSubmitText?: string;
+  pierOptions: Pier[];
+}
+
+const getBerthValidationSchema = (
+  t: TFunction,
+  pierOptions: Pier[]
+): ObjectSchema => {
+  return Yup.object().shape({
+    pierId: Yup.string()
+      .oneOf(pierOptions.map(pier => pier.id))
+      .required(t('forms.common.errors.required')),
+    number: Yup.number()
+      .typeError(t('forms.common.errors.numberType'))
+      .required(t('forms.common.errors.required')),
+    width: Yup.number()
+      .typeError(t('forms.common.errors.numberType'))
+      .required(t('forms.common.errors.required')),
+    length: Yup.number()
+      .typeError(t('forms.common.errors.numberType'))
+      .required(t('forms.common.errors.required')),
+    mooringType: Yup.string()
+      .oneOf(Object.keys(BerthMooringType))
+      .required(t('forms.common.errors.required')),
+  });
+};
+
+const BerthForm: React.FC<BerthFormProps> = ({
+  initialValues,
+  isSubmitting,
+  onSubmit,
+  onCancel,
+  onDelete,
+  onSubmitText,
+  pierOptions,
+}) => {
+  const { t } = useTranslation();
+  const validationSchema = getBerthValidationSchema(t, pierOptions);
+
+  const initial =
```

Comment:
Could this be written as follows to avoid the cast?
```suggestion
  const initial: Berth =
    initialValues ??
    ({
      pierId: pierOptions[0].id,
      mooringType: BerthMooringType.DINGHY_PLACE,
      comment: '',
      isActive: true,
    });
```

Reply:
Done.